### PR TITLE
cyphernetes 0.11.2 (new formula)

### PR DESCRIPTION
**Pull request by orphaninstance**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/cyphernetes.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  sha256 "2bb8945493178c7173b45dbeeac4c668d49d9010cab265185324d88567429549"
+  license "MIT"
+
+  depends_on "go" => :build
+  depends_on "make" => :build
+
+  def install
+    ENV["GOBIN"] = buildpath/"bin"
+    ENV.prepend_path "PATH", buildpath/"bin"
+    system "go", "install", "modernc.org/goyacc@v1.0.3"
```
Comment (SMillerDev) [inline Formula/c/cyphernetes.rb:14]: Why is this needed? Doesn't it just define this in the dependency file?

File: Formula/c/cyphernetes.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    system "make", "operator-manifests"
+    system "make", "build"
+
+    # The executable is actually built for the current environment, not necessarily darwin-arm64
+    mv("dist/cyphernetes-darwin-arm64", "dist/cyphernetes")
+    bin.install "dist/cyphernetes"
```
Comment (SMillerDev) [inline Formula/c/cyphernetes.rb:21]:
```suggestion
    bin.install "dist/cyphernetes-darwin-arm64" => "cyphernetes"
```

File: Formula/c/cyphernetes.rb (hunk 1 above)
Comment (anthonybrice) [reply Formula/c/cyphernetes.rb:14]: It's not a normal runtime dependency. `goyacc` is invoked from the shell at build time to generate a parser. As far as I know, I need the above to have it available on the path. I could be wrong however, I'm new to Go.

Comment (anthonybrice) [reply Formula/c/cyphernetes.rb:14]: @SMillerDev -- Upstream confirmed that this is intended. Please let me know if it's unacceptable, and we'll work on a resolution.

Comment (chenrui333) [reply Formula/c/cyphernetes.rb:14]: I am adding goyacc so that we can directly have it as build dep rather than having a "go install" step

**Review by chenrui333 — APPROVED** [review]

**Review by nandahkrishna — APPROVED** [review]

**Comment by AvitalTamir** [comment]
@anthonybrice @chenrui333 thanks for this contribution! 🙏
Please note a newer version (v0.11.1) has just been released.

**Comment by chenrui333** [comment]
Thanks @AvitalTamir, gonna update to the new release now.

File: Formula/c/cyphernetes.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  depends_on "goyacc" => :build
+  depends_on "make" => :build
+
+  def install
+    system "make", "operator-manifests"
+    system "make", "build"
+
+    # The executable is actually built for the current environment, not necessarily darwin-arm64
+    # upstream fix PR, https://github.com/AvitalTamir/cyphernetes/pull/84
+    bin.install "dist/cyphernetes-darwin-arm64" => "cyphernetes"
```
Comment (carlocab) [inline Formula/c/cyphernetes.rb:18]:
```suggestion
    system "make", "operator-manifests", "gen-parser"
    system "go", "build", *std_go_args(ldflags: "-s -w")
```

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]
